dns_server/src/main.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use clap::Parser;
+use config::DnsServerConfig;
+use crawler::{storage_impl::DnsServerStorageImpl, Crawler};
+use p2p::net::default_backend::{
+    transport::{NoiseEncryptionAdapter, NoiseTcpTransport, TcpTransportSocket},
+    DefaultNetworkingService,
+};
+use tokio::sync::mpsc;
+
+mod config;
+mod crawler;
+mod error;
+mod server;
+
+async fn run(config: Arc<DnsServerConfig>) -> Result<void::Void, error::DnsServerError> {
+    let (command_tx, command_rx) = mpsc::unbounded_channel();
+
+    let stream_adapter = NoiseEncryptionAdapter::gen_new();
+    let base_transport = TcpTransportSocket::new();
+    let transport = NoiseTcpTransport::new(stream_adapter, base_transport);
+    let chain_config = Arc::new(common::chain::config::create_mainnet());
+    let p2p_config = Default::default();
```

Comment:
Do we really have to instantiate a full Mintlayer node server? Can't we get away with simple pinging that port? Or just opening a connection and closing it?

Reply:
I'm using the generic `NetworkingService` from p2p, which works like a low-level networking implementation.
It's definitely not a full node server.
Simply pinging the ports is not enough, because we also need to know when new nodes join the network.